tests/alfacase/test_case_description.py

```diff
@@ -1054,3 +1054,103 @@ def test_invalid_fluid_reference_on_wells(default_well):
+
+    case = case_description.CaseDescription(
+        materials=[
+            case_description.MaterialDescription(name="Material"),
+            case_description.MaterialDescription(name="Material"),
+        ],
+        walls=[
+            case_description.WallDescription(name="Wall A"),
+            case_description.WallDescription(name="Wall A"),
+        ],
+        pvt_models=case_description.PvtModelsDescription(
+            default_model="PVT",
+            correlations={
+                "PVT1": case_description.PvtModelCorrelationDescription(),
+            },
+            compositions={
+                "PVT1": case_description.PvtModelCompositionalDescription(
+                    fluids={
+                        "Fluid 0": case_description.FluidDescription(),
+                        "Fluid 1": case_description.FluidDescription(),
+                    },
+                ),
+                "PVT2": case_description.PvtModelCompositionalDescription(
+                    fluids={
+                        "Fluid 0": case_description.FluidDescription(),
+                        "Fluid 1": case_description.FluidDescription(),
+                    }
+                ),
+            },
+        ),
+        nodes=[
+            case_description.NodeDescription(
+                name="ACME", node_type=NodeCellType.Pressure
+            ),
+            case_description.NodeDescription(
+                name="ACME", node_type=NodeCellType.Pressure
+            ),
+            case_description.NodeDescription(
+                name="Node1", node_type=NodeCellType.Pressure
+            ),
+            case_description.NodeDescription(
+                name="FOO", node_type=NodeCellType.Pressure
+            ),
+            case_description.NodeDescription(
+                name="FOO", node_type=NodeCellType.Pressure
+            ),
+        ],
+        pipes=[
+            case_description.PipeDescription(
+                name="Pipe 1",
+                source="ACME",
+                target="ACME",
+                segments=build_simple_segment(),
+            ),
+            case_description.PipeDescription(
+                name="Pipe 1",
+                source="ACME",
+                target="ACME",
+                segments=build_simple_segment(),
+            ),
+        ],
+        wells=[well_1, well_2],
+    )
+
+    expected_msg = dedent(
+        """\
+        Elements that can be referenced must have a unique name, found multiples definitions of the following items:
+        Nodes:
+            - ACME
+            - FOO
+        Pipes:
+            - Pipe 1
+        Wells:
+            - Well 1
+        Walls:
+            - Wall A
+        Materials:
+            - Material
+        PVT:
+            - PVT1
+        Fluids:
+            - Fluid 0
+            - Fluid 1"""
+    )
+    with pytest.raises(InvalidReferenceError, match=re.escape(expected_msg)):
+        case.ensure_unique_name()
```

Comment:
We also don't want:
- Pipes and Wells with same names; (because of OutputDefinition)
- Nodes and Wells with same names; (because of Edge source/target)

We should address those cases too (there may be others, but I can't remember other ones right now).

In the future, once we have _Equipment Trend_ in case description, edge's equipment will need to be unique accross the entire network, but since it is not available yet, you can leave that one out.